skrub/_agg_joiner.py

```diff
@@ -390,49 +434,115 @@ class AggTarget(TransformerMixin, BaseEstimator):
+    ...     operations=["mean", "max"],
     ... )
     >>> agg_target.fit_transform(X, y)
-       flightId  from_airport  ...  y_0_max_target y_0_mean_target
-    0         1             1  ...               1        0.666667
-    1         2             1  ...               1        0.500000
-    2         3             1  ...               1        0.500000
-    3         4             2  ...               1        0.666667
-    4         5             2  ...               1        0.666667
-    5         6             2  ...               1        1.000000
+       flightId  from_airport  ...  y_0_mean_target  y_0_max_target
+    0         1             1  ...         0.666667               1
+    1         2             1  ...         0.500000               1
+    2         3             1  ...         0.500000               1
+    3         4             2  ...         0.666667               1
+    4         5             2  ...         0.666667               1
+    5         6             2  ...         1.000000               1
     """
 
     def __init__(
         self,
         main_key,
-        operation=None,
-        suffix=None,
+        operations,
+        *,
+        suffix="_target",
     ):
         self.main_key = main_key
-        self.operation = operation
+        self.operations = operations
         self.suffix = suffix
 
+    def check_inputs(self, X, y):
+        """Check inputs before fitting.
+
+        Parameters
+        ----------
+        X : DataFrameLike
+            Input data, base table on which to left join the target.
+        y : DataFrameLike or SeriesLike or ArrayLike
+            The target to check.
+
+        Returns
+        -------
+        y_ : DataFrameLike
+            The transformed target.
+        """
+        if not sbd.is_dataframe(X):
+            raise TypeError(f"X must be a dataframe, got {type(X)}")
+
+        self._main_key = atleast_1d_or_none(self.main_key)
+        _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
+
+        # `y` is copied or converted to a df to be compatible with `aggregate`
+
+        # If `y` is already a dataframe
+        if sbd.is_dataframe(y):
+            y_ = sbd.make_dataframe_like(y, sbd.to_column_list(y))
+        # If `y` is a named series, we convert it to a dataframe
+        elif sbd.is_column(y) and sbd.name(y) is not None:
+            y_ = sbd.make_dataframe_like(y, {sbd.name(y): y})
+        else:
+            # If `y` is an unnamed series
+            if sbd.is_column(y):
+                y = sbd.to_numpy(y).reshape(-1, 1)
+            # If `y` is a list
+            elif isinstance(y, list):
+                y = np.atleast_2d(y)
+            # 1d array that needs to be reshaped
+            elif isinstance(y, np.ndarray) & len(y.shape) == 1:
+                y = y.reshape(-1, 1)
+
+            cols = {f"y_{i}": y[:, i] for i in range(y.shape[1])}
+            y_ = sbd.make_dataframe_like(X, cols)
```

Comment:
I guess we could use a single `y_ = sbd.make_dataframe_like(X, cols)` and have all previous conditions to only define `cols`, WDYT?

Btw, is there a typo here? Is this tested?
```suggestion
            y_ = sbd.make_dataframe_like(y, cols)
```

Reply:
the first argument of `make_dataframe_like` is only used for dispatch ie to choose the type of dataframe (pandas or polars), so either x or y would work as long as it comes from the correct module

Reply:
Ok, not what I would have expected TBH